src/main/java/domain/LottoGenerator.java

```diff
@@ -0,0 +1,56 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class LottoGenerator {
+	private static final int LOWER_BOUND = 1;
+	private static final int UPPER_BOUND = 45;
+	private static final int START_INDEX = 0;
+	private static final int END_INDEX = 6;
+	private static final String ERROR_NOT_IN_RANGE = "[ERROR] 1이상 45 이하의 숫자만 허용됩니다.";
+	private static final String ERROR_NOT_CORRECT_SIZE = "[ERROR] 지난 주 당첨 번호는 반드시 6개여야 합니다.";
+	private static final String ERROR_DUPLICATE_DETECT = "[ERROR] 지난 주 당첨 번호는 중복일 수 없습니다.";
+
+	private static final ArrayList<LottoNumber> candidates = (ArrayList<LottoNumber>) IntStream.range(LOWER_BOUND,
+		UPPER_BOUND + 1).mapToObj(LottoNumber::new).collect(Collectors.toList());
+
+	public static List<LottoNumber> generateRandomLottoNumbers() {
+		Collections.shuffle(candidates);
+		return new ArrayList<>(candidates.subList(START_INDEX, END_INDEX));
+	}
+
+	public static List<LottoNumber> generateAnswerLottoNumbers(List<Integer> userInput) {
+		validateSizeSix(userInput);
+		validateInRange(userInput);
+		validateDuplicate(userInput);
+		List<LottoNumber> lottoNumbers = new ArrayList<>();
+		for (int eachNum : userInput) {
+			lottoNumbers.add(new LottoNumber(eachNum));
+		}
+		return lottoNumbers;
+	}
+
+	private static void validateInRange(List<Integer> userInput) {
+		int count = (int) userInput.stream().filter(num -> num < LOWER_BOUND || num > UPPER_BOUND).count();
+		if (count > 0) {
+			throw new IllegalArgumentException(ERROR_NOT_IN_RANGE);
+		}
+	}
+
+	private static void validateSizeSix(List<Integer> userInput) {
+		if (userInput.size() != 6) {
+			throw new IllegalArgumentException(ERROR_NOT_CORRECT_SIZE);
+		}
+	}
+
+	private static void validateDuplicate(List<Integer> userInput) {
+		if (userInput.size() != new HashSet<Integer>(userInput).size()) {
```

Comment:
```suggestion
		if (userInput.size() != Set.of(userInput).size()) {
```

자바 11을 사용한다면 위와 같이 사용할 수 있을 것 같아요

Reply:
넵! 수정하였습니다

Reply:
말씀해주신 `Set.of(userInput)`으로 수정하게 되면 `[[1,2,3,4,5,6]]` 처럼 되어 size가 항상 1이 됩니다. 그렇게 되면 `useInput`의 올바른 입력에도 테스트에서 에러를 출력하게 됩니다. `Set`의 자료구조 특성상 index를 통한 접근이 불가능하기 때문에 제 방식이 목적에는 조금 더 부합하는 거 같은데 루피의 생각은 어떤지 궁금합니다!

Reply:
헉, 이것은 제가 코드를 잘못 본 것 같네요 ㅎㅎ 🙏